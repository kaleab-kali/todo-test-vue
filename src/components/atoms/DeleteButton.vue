<template>
  <button :class="$style.delete" data-cy="delete" @click="$emit('click')">
    &times;
  </button>
</template>

<script>
export default {};
</script>

<style lang="scss" module>
.delete {
  margin-right: 1.25rem;
  border: 1.5px solid $lightOrange;
  border-radius: 2px;
  color: $lightOrange;
  background-color: $darkPeach;
  max-height: 1.5rem;
}
.delete:active {
  color: red;
}
.delete:hover {
  border: 1.5px solid $orange;
  color: $orange;
  cursor: pointer;
}
</style>
