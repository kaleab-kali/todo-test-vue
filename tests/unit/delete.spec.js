import DeleteButton from "@/components/atoms/DeleteButton.vue";
import { mount } from "@vue/test-utils";

describe("Atom DeleteButton Component", () => {
  it("Correct layout", async () => {
    const wrapper = mount(DeleteButton);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("is clicked", async () => {
    const wrapper = mount(DeleteButton);
    await wrapper.find("button").trigger("click");
    expect(wrapper.emitted().click).toBeTruthy();
  });
});
