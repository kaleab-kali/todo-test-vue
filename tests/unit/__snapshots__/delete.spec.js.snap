// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom DeleteButton Component Correct layout 1`] = `<button class="delete">×</button>`;

exports[`Atom DeleteButton Component Correct layuot 1`] = `<button class="delete">×</button>`;

exports[`DeleteButton.vue is on click 1`] = `<button class="delete">×</button>`;
